File: test/testing_code.java
import java.io.File;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;
import java.util.Scanner;

public class Test {

    public static void main(String[] args) {
        // Insecure File Permissions
        File file = new File("sensitive_data.txt");
        file.setReadable(true, false);
        file.setWritable(true, false);

        // Rooted Device Access Simulation
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter device ID:");
        String deviceId = scanner.nextLine();
        System.out.println("Accessing device with ID: " + deviceId);

        // Lack of Hashing
        String data = "password123";
        try {
            MessageDigest md = MessageDigest.getInstance("MD5");
            byte[] hash = md.digest(data.getBytes());
            System.out.println("MD5 hash: " + new String(hash));
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
    }
}

File: test/testing_code.dart
import 'dart:io';
import 'dart:convert';
import 'dart:math';
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:crypto/crypto.dart';

void main() {
  // Insecure File Permissions Example
  File file = File('sensitive_data.txt');
  file.writeAsStringSync('Sensitive Data');

  // Rooted Device Access Simulation (Pseudo-code)
  // Note: In Flutter, detecting rooted devices requires platform-specific code, so this is a placeholder for illustration.
  print('Accessing device with ID: ROOTED_DEVICE_ID');

  // Lack of Hashing Example
  String data = 'password123';
  var bytes = utf8.encode(data);
  var digest = md5.convert(bytes);
  print('MD5 hash: ${digest.toString()}');

  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('Flutter Security Example'),
        ),
        body: Center(
          child: Text('Security vulnerabilities demo'),
        ),
      ),
    );
  }
}
